libpay-braintree/src/main/scala/com/wix/pay/braintree/JsonBraintreeOAuthMerchantParser.scala

```diff
@@ -0,0 +1,16 @@
+package com.wix.pay.braintree
+
+import org.json4s.DefaultFormats
+import org.json4s.native.Serialization
+
+class JsonBraintreeOAuthMerchantParser() extends BraintreeMerchantParser {
```

Comment:
@dleshem has just jumped behind me and suggested that parser will know which merchant it is (hints) so we dont need 2 parsers (so we wont need 2 gateways) 

will do that and update PR